rollup/internal/controller/sender/sender.go

```diff
@@ -608,6 +623,23 @@ func (s *Sender) checkPendingTransaction() {
 			}
 
 			if err := s.client.SendTransaction(s.ctx, newSignedTx); err != nil {
+				// SendTransaction failed, need to rollback the previous database changes
```

Comment:
Would it be clearer like this?

```go
if err := s.client.SendTransaction(s.ctx, newSignedTx); err != nil {
    tx.Rollback()
} else {
    tx.Commit()
}
```

Reply:
this cannot prevent the service is forcefully shutdown by k8s right after sending the tx but not updating db. which is found by altlayer.

Reply:
the fix pr of the bug found by `altlayer`'s testing: https://github.com/scroll-tech/scroll/pull/1564